yrs/src/block.rs

```diff
@@ -527,10 +567,9 @@ impl Item {
         if let Some(origin_id) = self.origin {
             let ptr = BlockPtr::from(origin_id);
             if let Some(item) = txn.store.blocks.get_item(&ptr) {
-                let id = self.origin.unwrap();
                 let len = item.len();
-                if id.clock == item.id.clock + len - 1 {
-                    self.left = Some(ptr);
+                if origin_id.clock == item.id.clock + len - 1 {
```

Comment:
`origin_id` and `id` point to the same block but are not necessarily equal - if block length is > 1, then these two can easily differ.